cloudferry_devlab/cloudferry_devlab/generate_load.py

```diff
@@ -265,15 +266,21 @@ def update_filtering_file(self):
             vm_id_list = [x.__dict__['id'] for x in
                           self.novaclient.servers.list()]
 
-            img_id_list = [x.__dict__['id'] for x in
-                           self.glanceclient.images.list(search_opts={
+            img_name_list = [x.__dict__['name'] for x in
+                             self.glanceclient.images.list(search_opts={
                                'is_public': False})]
 
             for key in filter_dict.keys():
                 if key == 'images':
-                    for img_id in img_id_list:
-                        if img_id not in not_incl_img:
-                            filter_dict[key]['images_list'].append(str(img_id))
+                    for img_name in img_name_list:
+                        if (img_list == 'exclude_images_list' and img_name
+                                in self.config.images_to_be_blacklisted) or \
+                                (img_list == 'images_list' and img_name not in
+                                 self.config.images_not_included_in_filter and
+                                 img_name not in
+                                 self.config.images_to_be_blacklisted):
+                            filter_dict[key][img_list].append(
```

Comment:
The filter file supports "images_list" section for included images and "exclude_images_list" section for blacklisted images. Both sections can't be filled together, which means if there are image ID filled in under "images_list" section, no image ID is allowed to be filled in under "exclude_images_list" section and vice versa is also true. This is the validation added in the functionality implemented in the cloudFerry for blacklisted images. I will optimize this logic.